### PR TITLE
[cuDNN][cuDNN RNNv8 API] Fix math type behavior in cuDNN RNN

### PR DESCRIPTION
**Pull request by eqy**
Adds back `CUDNN_TENSOR_OP_MATH` which was erroneously dropped by #115719 

CC @malfet @ptrblck 

cc @csarofeen @ptrblck @xwang233 @zou3519 @mikaylagawarecki

### CODE REVIEW

File: aten/src/ATen/cudnn/Descriptors.h [hunk 1]
```diff
@@ -318,6 +318,15 @@ struct TORCH_CUDA_CPP_API RNNDescriptor : public Descriptor<
       }
     }
 #else
+    cudaDeviceProp* prop = at::cuda::getCurrentDeviceProperties();
+    auto math_type = CUDNN_DEFAULT_MATH;
+    if (prop->major >= 7) {
+      if (input_type == CUDNN_DATA_HALF) {
+        math_type = CUDNN_TENSOR_OP_MATH;
+      } else if (allow_tf32) {
```
Comment (Shan19900305) [inline aten/src/ATen/cudnn/Descriptors.h:326]: Hi, This is from cudnn API doc, and " With NVIDIA Ampere architecture and CUDA toolkit 11, CUDNN_DEFAULT_MATH permits TF32 Tensor Core operation and CUDNN_FMA_MATH does not."
So when allow_tf32 == false, set math_type to CUDNN_FMA_MATH?

Comment (eqy) [reply aten/src/ATen/cudnn/Descriptors.h]: Nice catch! Yes that's a typo, it should be `!allow_tf32`

**Comment by drisspg** [comment]
https://github.com/pytorch/pytorch/blob/main/.ci/docker/common/install_cudnn.sh

I wonder what would the effort but to have a some newer cudnn builds in ci

**Review by drisspg — APPROVED** [review]

**Comment by eqy** [comment]
> https://github.com/pytorch/pytorch/blob/main/.ci/docker/common/install_cudnn.sh
> 
> I wonder what would the effort but to have a some newer cudnn builds in ci

The main effort would be tracking down all of the different paths (e.g., incl. windows) as attempted in #119710.

**Pull request closed without merging**